client/a8c-for-agencies/sections/referrals/lib/get-estimated-commission.ts

```diff
@@ -17,62 +17,82 @@ export const getDailyPrice = ( product: APIProductFamilyProduct, quantity: numbe
 
-			// Day the license was revoked if present
-			const revokedAt = purchase.license.revoked_at
-				? new Date( purchase.license.revoked_at )
-				: null;
+					// Day the license was issued
+					const issuedDate = new Date( purchase.license.issued_at );
+					// Set hours to 0 to compare from start of the day
+					issuedDate.setHours( 0, 0, 0, 0 );
 
-			// Start date is the latest of the license issued date and activity window start
-			const start = Math.max( issuedDate.getTime(), activityWindow.start.getTime() );
-			// Finish date is the earliest of the license revoked date and activity window finish
-			const finish = Math.min(
-				revokedAt ? revokedAt.getTime() : Infinity,
-				activityWindow.finish.getTime()
-			);
+					// Day the license was revoked if present
+					const revokedAt = purchase.license.revoked_at
+						? new Date( purchase.license.revoked_at )
+						: null;
 
-			// Total days is the difference between finish and start dates in days
-			// We add 1 to include end-to-end days
-			const totalDays = Math.floor( ( finish - start ) / ( 1000 * 60 * 60 * 24 ) ) + 1;
+					// Start date is the latest of the license issued date and activity window start
+					const start = Math.max( issuedDate.getTime(), activityWindow.start.getTime() );
+					// Finish date is the earliest of the license revoked date and activity window finish
+					const finish = Math.min(
+						revokedAt ? revokedAt.getTime() : Infinity,
+						activityWindow.finish.getTime()
+					);
 
-			if ( totalDays < 1 ) {
-				continue;
-			}
+					// Total days is the difference between finish and start dates in days
+					// We add 1 to include end-to-end days
+					const totalDays = Math.floor( ( finish - start ) / ( 1000 * 60 * 60 * 24 ) ) + 1;
 
-			const dailyPrice = getDailyPrice( product, purchase.quantity );
+					if ( totalDays < 1 ) {
+						continue;
+					}
 
-			// Get commission percentage for the product
-			const commissionPercentage = getProductCommissionPercentage( product.family_slug );
+					const dailyPrice = getDailyPrice( product, purchase.quantity );
 
-			// Calculate commission for the day
-			acc += dailyPrice * totalDays * commissionPercentage;
-		}
-		return acc;
-	}, 0 );
+					// Get commission percentage for the product (common for both subscription and license)
+					const commissionPercentage = getProductCommissionPercentage( product.family_slug );
+
+					// Add commission to the total commission (in cents)
+					acc.legacyCommissionsInCents += dailyPrice * totalDays * commissionPercentage;
+				}
+			}
+			return acc;
+		},
+		{ bdCommissions: 0, legacyCommissionsInCents: 0 }
+	);
 
-	// Convert commission to dollars and round
-	const totalCommission = Number( ( totalCommissionInCents / 100 ).toFixed( 2 ) );
+	// Convert commission from cents to dollars,
+	// add subscriptions commission and round to 2 decimal places
+	const totalCommission = Number( ( bdCommissions + legacyCommissionsInCents / 100 ).toFixed( 2 ) );
 
```

Comment:
I guess `bdCommissions` isn't in cents, right? If that's the case, I'd suggest wrapping the legacyCommissionsInCents / 100 part in parentheses to make the intention clearer and avoid confusion.

Reply:
Adding parentheses isn't possible because it is redundant and gets removed when the file is saved. So, I have moved the logic a bit to avoid the confusion: https://github.com/Automattic/wp-calypso/pull/104900/commits/a5168216ed23ca8e301fb4bfacad40be4769f466